src/sentry/eventstream/kafka/protocol.py

```diff
@@ -96,3 +98,71 @@ def get_task_kwargs_for_message(value):
         )
 
     return handler(*payload[1:])
+
+
+def get_task_kwargs_for_message_from_headers(headers: Sequence[Tuple[str, Optional[bytes]]]):
+    """
+    Same as get_task_kwargs_for_message but gets the required information from
+    the kafka message headers.
+    """
+    try:
+        header_data = defaultdict(lambda: None, {k: v for k, v in headers})
```

Comment:
Not sure I follow the choice of adding a defaultdict here. Can't you just access all entries with `get` so you get the None if the key is not present in the dictionary without having to copy it to a new structure.